pkg/cache/scheduler/fair_sharing.go

```diff
@@ -38,17 +39,81 @@ type dominantResourceShareNode interface {
 	hierarchicalResourceNode
 }
 
-// dominantResourceShare returns a value ranging from 0 to math.MaxInt,
+// DRS contains the DominantResourceShare for some
+// node, with convenence methods for precise comparison.
+type DRS struct {
+	fairWeight       *resource.Quantity
+	unweightedRatio  float64
+	dominantResource corev1.ResourceName
+}
+
+// NegativeDRS is used as a starting point for comparisons.
+func NegativeDRS() DRS {
+	return DRS{unweightedRatio: -1, dominantResource: "", fairWeight: &oneQuantity}
+}
+
+// IsZero returns whether the DRS is 0. In other words,
+// the node for which this function was called is not
+// borrowing any resources.
+func (d DRS) IsZero() bool {
+	return d.unweightedRatio == 0
+}
+
+func (d DRS) PreciseWeightedShare() float64 {
+	if d.IsZero() {
+		return 0.0
+	}
+	if d.fairWeight.IsZero() {
+		return math.Inf(1)
+	}
+	return float64(d.unweightedRatio) * 1000.0 / float64(d.fairWeight.MilliValue())
```

Comment:
I see. So... MilliValue seems to collapse at `0.0001` based off https://github.com/kubernetes-sigs/kueue/issues/4247#issuecomment-3312129775

And resource.Quantity collapses at `1.0000000000000002e-07` (is that correct?) This one is okay for drs value. 

For weight precision, I can check in our clusters how many weights we have that go up to `0.0001`. Looks okay for now!

Reply:
I changed `MilliValue` to `AsFloat64Slow`, so we at least represent FairSharing weight down to `10^-6`, rather than just  `10^-3`